### ISSUE TITLE
AnyOfConstraint only working with arrays

### ISSUE THREAD

**Issue by ghost** [nunit/nunit#3633]
Consider the following test:
```
        [Test]
        public void AnyOf_OnlyWorksWithArray()
        {
            var expectedString = "Employee";
            var actualStrings = new List<string> { "Employee", "Manager", "Client" };
            
            var expectedNumber = 1;
            var actualNumbers = new List<int> { 1, 2, 3, 4, 5 };

            Assert.That(expectedString, Is.AnyOf(actualStrings));
            Assert.That(expectedNumber, Is.AnyOf(actualNumbers));
        }
```
It fails with the following message:
Message:   Expected: any of < < "Employee", "Client", "Security" > >
  But was:  "Employee"

The second assertion also fails, when asserted first:
Message:   Expected: any of < < 1, 2, 3, 4, 5 > >
  But was:  1

I was surprised to find out that Is.AnyOf() only works with arrays.

**Comment by mikkelbu** [comment]
I just noticed this issue and have transferred it to the framework repo.

**Comment by h0s** [comment]
@CharliePoole  i suggest as a solution adding another method `AnyOf` that accept list<object> and inside it there is a mechanism where we transfer it to array and we call the same `AnyOfConstraint`, 
what do you think about this ?

**Comment by stevenaw** [comment]
@h0s You may want to consider tagging @nunit/framework-team with your question, as you'd be able to get a wider range of opinions and inputs from the people on that list.

I've looked into the history of this constraint a bit in https://github.com/nunit/nunit/issues/2704 and it sounds like there was a bit of discussion over some of these edge cases around handling `List<>` and `string`. At the time, it sounded like an Analyzer was a preferred approach over adding an overload, but I'm unsure if any opinions have changed since then.

**Comment by undron** [comment]
After more than 7 years of using NUnit, I have only had one use for this constraint, but faced with this issue.
Would it be possible to simply add an overload that would accept IEnumerable<object>?

**Comment by mikkelbu** [comment]
Closed as duplicate of #4355 which was just solved by #4356 for v4
